### PR TITLE
Pseudo Terminal for extension taskManager

### LINKED ISSUE #190: [Bug Report]: Conflict with PlatformIO and ESP-IDF Extension (VSC-456)
**Describe the bug**
With the latest Visual Studio Code ESP-IDF extension installed, and PlatformIO installed, Visual Studio Code does not honor the path settings that the Visual Studio Code ESP-IDF extension attempts to set.  Notably, the first error that pops up during a build is the following:

'cmake' is not recognized as an internal or external command, operable program or batch file.

**To Reproduce**
1.) Install Visual Studio Code.  
2.) Install the ESP-IDF extension
3.) Install the PlatformIO extension
4.) Attempt to build with the ESP-IDF extension
(Note: the order of 2+3 doesn't matter)

**Expected behavior**
I'd expect that regardless of any other plugins, Visual Studio Code's ESP-IDF would build.  However, I'm not sure if the behavior described in this bug report is due to an issue in PlatformIO, or due to an issue with the ESP-IDF extension. 

**Environment (please complete the following information):**
Windows 7
VSCode 1.49.3
ESP-IDF Version 4.2-release 
VSCode ESP-IDF Extension 0.5.1
PlatformIO IDE 2.1.0
Python Version 3.8.6

**Logs**
'cmake' is not recognized as an internal or external command, operable program or batch file. The terminal process "C:\Windows\system32\cmd.exe /d /c cmake -G Ninja .." failed to launch (exit code: 1).

**Additional context**
I executed within a shell the ESP-IDF\export.bat and launched Visual Studio Code from that path, it solved the issue since it started Visual Studio Code with the correct paths to the tools. This firmly places the issue with the environment variable settings.

### PR DESCRIPTION
**Pull request by brianignacio5**
Use vscode.PseudoTerminal to implement this extension tasks to fix environment variables issues such as the conflict with the PlatformIO extension.

Should fix #190

### CODE REVIEW

File: src/espIdfCustomTerminal.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    const errHandler = (error: Error) => {
+      sendOutput(Buffer.from(error.message));
+      Logger.error(error.message, new Error(error.message));
+    };
+    return new Promise((resolve) => {
+      this.childProcess = spawn(cmd, args, this.options);
+      this.childProcess.stdout.on("data", sendOutput);
+      this.childProcess.stderr.on("data", sendOutput);
+      this.childProcess.on("error", errHandler);
+      this.childProcess.on("exit", (code) => {
```
Comment (pwmb) [inline src/espIdfCustomTerminal.ts:46]: What about `stdin` will it not be piped to the process from the custom terminal?

Comment (brianignacio5) [reply src/espIdfCustomTerminal.ts:46]: Well yes, but it is slightly different approach and just wanted to complete MVP for extension tasks to be fixed.

Was thinking of working this in another PR, or should put here ?

Comment (brianignacio5) [reply src/espIdfCustomTerminal.ts]: The thing this pseudo terminal is not a complete terminal.

vscode.PseudoTerminal has a `handleInput` method to manage these. I added an initial implementation to allow user interaction, but is limited due to each line input is an independent spawn process.

File: src/extension.ts [hunk 2]
```diff
@@ -1930,30 +1931,73 @@ function createIdfTerminal() {
...
+function createIdfCustomTerminal() {
+  PreCheck.perform([webIdeCheck, openFolderCheck], () => {
+    const modifiedEnv = utils.appendIdfAndToolsToPath();
+    const espCustomTerminal = new EspIdfCustomTerminal("", undefined, {
+      env: modifiedEnv,
+      cwd: workspaceRoot.fsPath || modifiedEnv.IDF_PATH || process.cwd(),
+      shell: vscode.env.shell,
+    });
+    const espCustomIdfTerminal = vscode.window.createTerminal({
+      name: "ESP-IDF Terminal",
```
Comment (pwmb) [inline src/extension.ts:1947]: Maybe ESP-IDF custom terminal

**Review by pwmb — COMMENTED** [review]
Left a few comments, other than that LGTM!

File: README.md [hunk 3]
```diff
@@ -111,6 +111,7 @@ Click <kbd>F1</kbd> to show Visual studio code actions, then type **ESP-IDF** to
 | Monitor your device                             | <kbd>⌘</kbd> <kbd>E</kbd> <kbd>M</kbd> | <kbd>Ctrl</kbd> <kbd>E</kbd> <kbd>M</kbd> |
 | Build, Flash and start a monitor on your device | <kbd>⌘</kbd> <kbd>E</kbd> <kbd>D</kbd> | <kbd>Ctrl</kbd> <kbd>E</kbd> <kbd>D</kbd> |
 | Open ESP-IDF Terminal                           | <kbd>⌘</kbd> <kbd>E</kbd> <kbd>T</kbd> | <kbd>Ctrl</kbd> <kbd>E</kbd> <kbd>T</kbd> |
+| Open ESP-IDF Custom Terminal                    |                                        |                                           |
```
[the remark this answers is not in the document]
Comment (brianignacio5) [reply README.md:114]: Just left it so others can test the terminal, but it is only for internal use of the extension. Will remove

[the remark this answers is not in the document]
Comment (pwmb) [reply README.md]: Also add how user can switch default shell for this.

**Review by pwmb — CHANGES_REQUESTED** [review]

**Pull request merged**